pkg/querier/queryrange/codec.go

```diff
@@ -173,38 +173,74 @@ func (r *LokiSeriesRequest) LogToSpan(sp opentracing.Span) {
 
 func (*LokiSeriesRequest) GetCachingOptions() (res queryrangebase.CachingOptions) { return }
 
-func (r *LokiLabelNamesRequest) GetEnd() int64 {
-	return r.EndTs.UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+// In some other world LabelRequest could implement queryrangebase.Request.
+type LabelRequest logproto.LabelRequest
```

Comment:
Unfortunately our type for times are not unified. And we cannot change `logproto.LabelRequest` because it might change the API.